packages/internal/src/upgrade-api.js

```diff
@@ -1,18 +1,34 @@
 // @ts-check
 // @jessie-check
-import { isObject } from '@endo/marshal';
+
+import { M, matches } from '@endo/patterns';
+
+const { isFrozen } = Object;
 
 /**
- * @typedef {{ name: string, upgradeMessage: string, incarnationNumber: number }} DisconnectionObject
+ * An Error-like object for use as the rejection reason of promises
+ * abandoned by upgrade.
+ *
+ * @typedef {{
+ *   name: string,
```

Comment:
Should this be narrowed?
```suggestion
 *   name: 'vatUpgraded',
```

Reply:
Good. Done.